mapie/calibration.py

```diff
@@ -0,0 +1,502 @@
+from typing import Optional, Union, cast, Tuple, Dict
+
+import warnings
+import numpy as np
+from sklearn.base import BaseEstimator, ClassifierMixin, clone, RegressorMixin
+from sklearn.calibration import _SigmoidCalibration
+from sklearn.isotonic import IsotonicRegression
+from sklearn.pipeline import Pipeline
+from sklearn.model_selection import BaseCrossValidator
+from sklearn.utils import check_random_state, check_consistent_length
+from sklearn.utils.multiclass import type_of_target
+from sklearn.utils.validation import (
+    indexable,
+    check_is_fitted,
+    _check_y,
+    _num_samples,
+)
+
+from ._typing import ArrayLike, NDArray
+from .utils import (
+    check_cv,
+    check_null_weight,
+    check_n_features_in,
+    check_calib_set,
+    fit_estimator,
+    check_estimator_classification,
+    check_estimator_fit_predict,
+    check_binary_zero_one,
+)
+
+
+class MapieCalibrator(BaseEstimator, ClassifierMixin):
+
+    fit_attributes = [
```

Comment:
Please add a docstring with the arguments of the init + a small theoretical description + a small example and the references of the paper you used. Take example on classification.py

Reply:
done.